rule.go

```diff
@@ -349,15 +353,19 @@ func (r *Rule) Evaluate(tx *Transaction) []MatchData {
 
 	// disruptive actions are only evaluated by parent rules
 	if r.ParentID == 0 {
+		matchedChainValues := matchedValues
```

Comment:
Maybe ```matchedChainValues := []MatchData{}``` is more appropriate.

Reply:
Have moved this in inner loop to preserve intermediate matchedChainValues